src/framework/mocha/MochaRuntime.m

```diff
@@ -1003,10 +1003,18 @@ - (void)finalizeForMochaScript {
 #pragma mark -
 #pragma mark Global Object
 
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
+NSSet* JS_GLOBAL_OBJECTS;
+
 JSValueRef Mocha_getProperty(JSContextRef ctx, JSObjectRef object, JSStringRef propertyNameJS, JSValueRef *exception) {
+    
+    if (JS_GLOBAL_OBJECTS == nil) {
+        JS_GLOBAL_OBJECTS = [NSSet setWithObjects: @"Infinity", @"NaN", @"undefined", @"null", @"eval", @"isFinite", @"isNaN", @"parseFloat", @"parseInt", @"decodeURI", @"encodeURI", @"decodeURIComponents", @"encodeURIComponents", @"Object", @"Function", @"Boolean", @"Symbol", @"Error", @"EvalError", @"InternalError", @"RangeError", @"ReferenceError", @"SyntaxError", @"TypeError", @"URIError", @"Number", @"Math", @"Date", @"String", @"RegExp", @"Array", @"Int8Array", @"Uint8Array", @"Uint8ClampedArray", @"Int16Array", @"Uint16Array", @"Int32Array", @"Uint32Array", @"Float32Array", @"Float64Array", @"Map", @"Set", @"WeakMap", @"WeapSet", @"SIMD", @"ArrayBuffer", @"DataView", @"JSON", @"Promise", @"GeneratorFunction", @"AsyncFunction", @"Refect", @"Proxy", @"Intl", @"WebAssembly", nil];
```

Comment:
- Typo: According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects, `decodeURIComponents` should be `decodeURIComponent`, and `encodeURIComponents` should be `encodeURIComponent`.
- Typo: Ditto for `WeapSet` → `WeakSet`
- `Refect` → `Reflect`
- `Generator` seems to be missing
- `arguments` is also missing from this list, not sure if that's intentional or maybe it's already defined somewhere else?